src/nodes/MemberAccess.js

```diff
@@ -4,15 +4,42 @@ const {
   }
 } = require('prettier/standalone');
 
-const isBeginnigOfChain = (path) => {
-  const parentNodeType = path.getParentNode().type;
+const isEndOfChain = (node, path) => {
+  let i = 0;
+  let currentNode = node;
+  let parentNode = path.getParentNode(i);
+  while (parentNode) {
+    // If direct ParentNode is a MemberAcces we are not at the end of the chain
+    if (parentNode.type === 'MemberAccess') return false;
 
-  if (parentNodeType === 'MemberAccess') return false;
-  if (parentNodeType === 'FunctionCall') {
-    const grandParentNodeType = path.getParentNode(1).type;
-    return grandParentNodeType !== 'MemberAccess';
-  }
+    // If direct ParentNode is a FunctionCall and currentNode is not the expression
+    // then it must be and argument in which case it is the end of the chain.
+    if (
+      parentNode.type === 'FunctionCall' &&
+      currentNode !== parentNode.expression
+    )
+      break;
```

Comment:
can we use return instead of break?

Reply:
If we return, we might get to a point where we add all of the possible scenarios and the last return never runs (but we need to keep it for safety) thus breaking our 100% streak.